### ISSUE TITLE
Enhance limited evaluation of setup.py parsing: skip on fail

### ISSUE THREAD

**Issue by mknorps** [tweag/FawltyDeps#174]
Testing FawltyDeps on real project: [pyfolio](https://github.com/quantopian/pyfolio/), I encountered following setup.py structure:
```python
...
support_ipython_6 = (sys.version_info >= (3, 3))

install_reqs = [
    'ipython>=3.2.3' if support_ipython_6 else 'ipython>=3.2.3, <6',
    'matplotlib>=1.4.0',
    'numpy>=1.11.1',
    'pandas>=0.18.1',
    'pytz>=2014.10',
    'scipy>=0.14.0',
    'scikit-learn>=0.16.1',
    'seaborn>=0.7.1',
    'empyrical>=0.5.0',
]

test_reqs = ['nose>=1.3.7', 'nose-parameterized>=0.5.0', 'runipy>=0.1.3']
...
if __name__ == "__main__":
    setup(
...
        install_requires=install_reqs,
        extras_require=extras_reqs,
        tests_require=test_reqs,
        test_suite='nose.collector',
    )
```

FD fails on line:
```
    'ipython>=3.2.3' if support_ipython_6 else 'ipython>=3.2.3, <6',
```
because it cannot evaluate that.
If we skip (and report) the line we cannot evaluate in a list like that instead of skipping the whole list, we could extend our accuracy.

**Comment by jherland** [comment]
(using this issue to collect more instances where more advanced setup.py parsing is needed)

dparse is a project with similar issues to this, and their https://github.com/pyupio/dparse/issues/2 points to this setup.py as an example of one that is hard to parse correctly: https://github.com/httpie/cli/blob/f1d4861faed6486a356175ce9f92d4da96ddba01/setup.py
